Improve scaling performance of xg

ekg opened the pull request:
A number of data structures in xg (version 3) have non-constant time with respect to the index scale. I'm going to explore such effects in this branch and merge if I can convince myself that the time/space tradeoff tilts in favor of uncompressed versions of these structures.

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/733/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/083e3207b984820490a23d3cae25e42c3b951cc4/index.html).

10 tests passed, 0 tests failed and 10 tests skipped in 9391 seconds

Tests produced 2156 warnings. 2156 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/741/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/15dd95d735ad0be15153c2b9811d2f9b11b0d2df/index.html).

10 tests passed, 0 tests failed and 11 tests skipped in 9777 seconds

Tests produced 1974 warnings. 1969 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/746/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/dc1eda64a5b1ea1452c679f7bcffd0823522fc56/index.html).

9 tests passed, 1 tests failed and 11 tests skipped in 10489 seconds

Failed tests:

* test_sim_chr21_snp1kg (4936 seconds)

Tests produced 6 warnings. 6 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/760/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/ca44d0c50bccb2b9170b762c912e53e08f34d4bc/index.html).

10 tests passed, 0 tests failed and 11 tests skipped in 9715 seconds

Tests produced 1972 warnings. 1972 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/761/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/f85885825416e9487fbe32eac76e3b87803b0c60/index.html).

9 tests passed, 1 tests failed and 11 tests skipped in 9376 seconds

Failed tests:

* test_map_brca1_snp1kg (55 seconds)

Tests produced 1962 warnings. 1962 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/763/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/099a809d325f4f62141855d36dc8a3812116f1c5/index.html).

9 tests passed, 1 tests failed and 11 tests skipped in 9358 seconds

Failed tests:

* test_map_brca1_snp1kg (56 seconds)

Tests produced 1963 warnings. 1963 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/764/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/38a43b1188a31571d2405c27d956c97fc5c35a7b/index.html).

9 tests passed, 1 tests failed and 11 tests skipped in 9912 seconds

Failed tests:

* test_map_brca1_snp1kg (55 seconds)

Tests produced 1959 warnings. 1959 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/765/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/1a894dea12712ca98f282fd673fb4179d4e08857/index.html).

9 tests passed, 1 tests failed and 11 tests skipped in 9616 seconds

Failed tests:

* test_map_brca1_snp1kg (56 seconds)

Tests produced 1961 warnings. 1961 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/767/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/ab9a689bbf275f8d20d2b5ff7e9c851c1f90db40/index.html).

1 tests passed, 9 tests failed and 11 tests skipped in 6200 seconds

Failed tests:

* test_full_brca2_cactus (261 seconds)
* test_full_brca2_primary (237 seconds)
* test_full_brca2_snp1kg (174 seconds)
* test_map_brca1_cactus (38 seconds)
* test_map_brca1_primary (19 seconds)
* test_map_brca1_snp1kg (28 seconds)
* test_map_mhc_primary (86 seconds)
* test_map_mhc_snp1kg (28 seconds)
* test_sim_chr21_snp1kg (3447 seconds)

Tests produced 7 warnings. 7 were for lower-than-expected alignment scores

glennhickey commented:
Jenkins, test this please.

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/769/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/ab9a689bbf275f8d20d2b5ff7e9c851c1f90db40/index.html).

9 tests passed, 1 tests failed and 11 tests skipped in 9912 seconds

Failed tests:

* test_map_brca1_snp1kg (56 seconds)

Tests produced 1958 warnings. 1958 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/777/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/a1f4d1560eb2105e4557976818d0aa98e658d6ed/index.html).

9 tests passed, 1 tests failed and 10 tests skipped in 8945 seconds

Failed tests:

* test_map_brca1_snp1kg (56 seconds)

Tests produced 2131 warnings. 2131 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/786/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/6b76137820c6b41d9a2e96eacb9c6373949ed5ca/index.html).

9 tests passed, 1 tests failed and 10 tests skipped in 9142 seconds

Failed tests:

* test_map_brca1_snp1kg (58 seconds)

Tests produced 2134 warnings. 2134 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/788/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/a1a115d982a8e867f07dabaedf5227b0da6d162b/index.html).

9 tests passed, 1 tests failed and 10 tests skipped in 8947 seconds

Failed tests:

* test_map_brca1_snp1kg (57 seconds)

Tests produced 2136 warnings. 2136 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/799/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/9b75fd02a34ff8faca980868f97e4c4b8b0c2953/index.html).

9 tests passed, 1 tests failed and 10 tests skipped in 9161 seconds

Failed tests:

* test_map_brca1_snp1kg (57 seconds)

Tests produced 2130 warnings. 2130 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/804/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/28b4abdcf1a989312dff5a77fb185352599fb204/index.html).

13 tests passed, 0 tests failed and 10 tests skipped in 12636 seconds

Tests produced 4976 warnings. 4972 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/820/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/c731d1858fc20eb4f3304b9c27d025b997fde6ef/index.html).

13 tests passed, 0 tests failed and 10 tests skipped in 11304 seconds

Tests produced 4976 warnings. 4976 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/826/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/b3bb82a43b0dddeae854da177b5006f502ac931a/index.html).

13 tests passed, 0 tests failed and 10 tests skipped in 11234 seconds

Tests produced 4974 warnings. 4974 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/828/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/509fba32c3239b86ba843f131b81ac90b4c5cee1/index.html).

13 tests passed, 0 tests failed and 10 tests skipped in 11091 seconds

Tests produced 4955 warnings. 4955 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/830/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/de1be96988e8383d48c4e2fb25e0e317dc068ecf/index.html).

12 tests passed, 0 tests failed and 11 tests skipped in 10972 seconds

Tests produced 4995 warnings. 4995 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/835/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/eabca1c156bb594815315b937b8775c052a22c88/index.html).

12 tests passed, 0 tests failed and 11 tests skipped in 11636 seconds

Tests produced 4870 warnings. 4870 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/838/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/a40011f041bf3d23d33e61cbc8e123410a15920a/index.html).

12 tests passed, 0 tests failed and 11 tests skipped in 10668 seconds

Tests produced 5005 warnings. 5005 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/842/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/ac39af9888386daddca2a0f589c997c80ed066a1/index.html).

12 tests passed, 0 tests failed and 11 tests skipped in 11636 seconds

Tests produced 4847 warnings. 4847 were for lower-than-expected alignment scores

jeizenga commented:
I just checked the Jenkins for this last commit, and it looks like mpmap's performance is degrading. Are you changing the MEM code, maybe? If your changes end up hurting the mpmap performance in the final version you want to merge, can you maintain the current behavior as a separate code path so I can keep using it?

ekg commented:
@jeizenga how is b3bb82a4? I was going to revert to that before merging.

ekg commented:
The performance hit was certainly due to a change in the MEM algorithm. I realized after a review that too much has changed to fully revert. I implemented some bug fixes and a performance improvement that _should_ have no effect on the MEM finding. However, I'd forgotten to hook up the other options in the MEM finding, so we were always going down the find_sub_mems_fast route. I had rewritten the MEM finding to happen all at once at the end, in hopes that this might improve our performance when calling gcsa's count function due to improved caching. The rewrite seems to have caused something worse--- a genuine bug in the call to find_sub_mems_fast. Hopefully fixing this and hooking back in the other options will recover the performance at b3bb82a4. If it does, I'll move to merge.

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/844/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/8c88810050dfd61d0cef82a68b94763e2f210699/index.html).

11 tests passed, 1 tests failed and 11 tests skipped in 9190 seconds

Failed tests:

* test_map_brca1_snp1kg (47 seconds)

Tests produced 4726 warnings. 4726 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/863/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/e115742224e08e916fcace6ab11d4fd03c12752b/index.html).

12 tests passed, 1 tests failed and 10 tests skipped in 10379 seconds

Failed tests:

* test_sim_mhc_snp1kg (1583 seconds)

Tests produced 4724 warnings. 4724 were for lower-than-expected alignment scores

ekg commented:
How can I ask Jenkins to run this again?

adamnovak commented:
You say "Jenkins, test this please".

On Wed, Oct 18, 2017 at 2:01 PM, Erik Garrison <notifications@github.com>
wrote:

> How can I ask Jenkins to run this again?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vgteam/vg/pull/1135#issuecomment-337726657>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AE0_X6nevW_Uc1qaZi2E7079F7jwevPGks5stmcygaJpZM4PtyMz>
> .
>

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/869/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/79cbdb070a3177ba7622613044d8b29d3cabc1dc/index.html).

13 tests passed, 0 tests failed and 10 tests skipped in 9167 seconds

Tests produced 4894 warnings. 4894 were for lower-than-expected alignment scores

benedictpaten commented:
Also, might be good to merge in master.

On Wed, Oct 18, 2017 at 5:01 PM, Erik Garrison <notifications@github.com>
wrote:

> How can I ask Jenkins to run this again?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vgteam/vg/pull/1135#issuecomment-337726657>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAkOICmJZlVUv49_xVm1K1IrhjzYEo_Tks5stmcxgaJpZM4PtyMz>
> .
>



-- 
Benedict
(calendar invites: bpaten@ucsc.edu,
 appointments: Tina Bernard <tibernar@ucsc.edu>)

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/873/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/08f2f0332b5465068c90293741c239e22503735a/index.html).

13 tests passed, 0 tests failed and 10 tests skipped in 9402 seconds

Tests produced 5033 warnings. 5033 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/874/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/b2564c4a2b3df3043a735424512c99db52260087/index.html).

13 tests passed, 0 tests failed and 10 tests skipped in 9229 seconds

Tests produced 5061 warnings. 5061 were for lower-than-expected alignment scores

cgcloud-jenkins commented:
[Jenkins vg tests](http://jenkins.cgcloud.info/job/vg/880/) complete for [PR 1135](https://github.com/vgteam/vg/pull/1135).  View the [full report here](http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/3300c61bf86e3e2d44545503d08f78faeab3c2be/index.html).

13 tests passed, 0 tests failed and 10 tests skipped in 9090 seconds

Tests produced 5039 warnings. 5039 were for lower-than-expected alignment scores
